core/src/main/java/org/apache/accumulo/core/metadata/schema/TabletsMetadata.java

```diff
@@ -87,13 +91,20 @@
     private TableId tableId;
     private ReadConsistency readConsistency = ReadConsistency.IMMEDIATE;
     private AccumuloClient _client;
+    private Collection<KeyExtent> extents;
 
     Builder(AccumuloClient client) {
       this._client = client;
     }
 
     @Override
     public TabletsMetadata build() {
+      if (extents != null) {
+        Preconditions.checkState(
+            range == null && table == null && level == DataLevel.USER && !checkConsistency);
+        return buildExtents(_client);
+      }
```

Comment:
This is doing a lot of state checking, but it's not clear what it's checking for. A comment could go a long way here.

Reply:
adde comment in 52b75b1